framework/src/media/MediaWorker.cpp

```diff
@@ -59,9 +59,34 @@ void MediaWorker::stopWorker()
 	}
 }
 
-MediaQueue& MediaWorker::getQueue()
+std::function<void()> MediaWorker::deQueue()
 {
-	return mWorkerQueue;
+	return mWorkerQueue.deQueue();
+}
+
+bool MediaWorker::processLoop()
+{
+	if (mWorkerQueue.isEmpty()) {
+		return true;
+	} else {
+		return false;
+	}
+}
+
+int MediaWorker::mediaLooper()
+{
+	medvdbg("MediaWorker : mediaLooper\n");
+
+	while (mIsRunning) {
+		while (processLoop());
```

Comment:
else not neccessary in processLoop and let's remain TODO what you mentioned about queue.